### ISSUE TITLE
Other optimisation methods like LBFGS?

### ISSUE THREAD

**Issue by YinghaoHuang91** [geomstats/geomstats#789]
Thanks for sharing this great project! Just wonder does it currently support other common optimisers like LFBGS or Conjugate Gradient methods? If so is there an example to follow?

**Comment by ninamiolane** [comment]
Thank you for your interest! We do not support many optimisers, as they are dedicated libraries for optimization on manifolds: `pymanopt`, `geoopt` for example. Geomstats focuses on differential geometry and learning. You can find [here](https://github.com/geomstats/geomstats/blob/master/examples/geomstats_in_pymanopt.py) an example using `pymanopt`'s solvers with `geomstats`' geometry.

https://github.com/geomstats/geomstats/blob/8c949af0c89b439dceb0ee8f17481efa5baed593/examples/geomstats_in_pymanopt.py#L83

Modify this line to get another solver from `pymanopt`, they do have conjugate gradient methods implemented: `solver = ConjugateGradient()`.

**Comment by YinghaoHuang91** [comment]
Thanks for your prompt response! I wonder is the gradient provided by geomstats is directly usable in other libraries like PyTorch and Tensorflow or not? Namely geomstats supports backends like Tensroflow or PyTorch, is the other way around also possible? If so is there an example I can refer to?

**Comment by ninamiolane** [comment]
Yes, see for example:
https://github.com/geomstats/applications/tree/master/se3_pose_estimation
which is the codefor this paper: https://arxiv.org/abs/1805.01026.

Note however that it relies on an old version of geomstats (v1.7), whose API has changed a lot since then.

**Comment by YinghaoHuang91** [comment]
That's great! Thanks a lot for your fast and patient response!
